src/lib/NavigationHub/Link/Route.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace EzSystems\HybridPlatformUi\NavigationHub\Link;
+
+use EzSystems\HybridPlatformUi\NavigationHub\Link;
+use Symfony\Component\HttpFoundation\Request;
+
+class Route extends Link
+{
+    protected $router;
+
+    protected $routeName;
+
+    protected $routeParams;
+
+    public function __construct($router, $routeName, $routeParams, $name, $zoneIdentifier)
```

Comment:
Typehint for `$router` (which should be renamed to `$urlGenerator`) against `UrlGeneratorInterface`